script.js

```diff
@@ -1 +1,402 @@
+      gameStatus = 'stopGame';
+      numOfGames += 1;
+      gameInitNextGame(makeDeck, shuffleCards, buildBoardElements);
+    }
+  }, 1000);
+
+  return matchTimerId;
+};
+
+// to initilize the next game when game timer runs out
+const gameInitNextGame = (makeDeck, shuffleCards, buildBoardElements) => {
+  // clear the browser content
+  document.body.innerHTML = '';
+
+  // empty the current board
+  board = [];
+
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck(); // must not shuffle this deck bc you want to take out the pairs
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the new board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      board[i].push(deck.pop());
+    }
+  }
+
+  const boardEl = buildBoardElements(board);
+
+  document.body.appendChild(boardEl);
+
+  // start timer till match ends
+  secondsLeftTillGameOver = 180;
+
+  timerId = startMatchTimer(gameInitNextGame, makeDeck, shuffleCards, buildBoardElements);
+
+  // initialize game info div with lose message
+  gameInfo.innerText = loseMessage;
+
+  document.body.appendChild(gameInfo);
+
+  document.body.appendChild(scoreMessage);
+
+  // display match's countdown timer
+  matchCountdownTimer.innerText = `Time Left: ${secondsLeftTillGameOver}s`;
+  document.body.appendChild(matchCountdownTimer);
+
+  // change global variables accordingly -------------------
+  gameStatus = 'playGame';
+  canClick = true;
+  numOfMatches = 0;
+};
+
+// output message
+const output = (message) => {
+  gameInfo.innerHTML = message;
+};
+
+// flip over cards that were clicked this round if they do not match
+const flipOverSquares = (firstSquare, secondSquare) => {
+  canClick = false;
+  setTimeout(() => {
+    console.log('turn both cards back over');
+    firstSquare.innerText = '';
+    secondSquare.innerText = '';
+
+    // allow player to click on squares again
+    canClick = true;
+  }, 3000);
+};
+
+// create match message that will disappear after a set time
+const createMatchMessage = (boardElement) => {
+  const matchMessageContainer = document.createElement('div');
+  matchMessageContainer.classList.add('match-message-container');
+  matchMessageContainer.innerText = 'MATCH!';
+
+  boardElement.appendChild(matchMessageContainer);
+
+  canClick = false;
+
+  // match message will disappear after a set time
+  setTimeout(() => {
+    boardElement.removeChild(matchMessageContainer);
+
+    canClick = true;
+
+    // tell player to click on another square
+    output('It is a match! <br> Click on another square to continue!');
+  }, 3000);
+};
+
+// flip cards and check for pairs when a square is clicked
+const squareClick = (squareClickElement, column, row, boardElement) => {
```

Comment:
This function is getting a bit long. Consider decomposing the logic in the conditionals into helper functions.